helpdesk_mgmt/controllers/main.py

```diff
@@ -0,0 +1,77 @@
+import base64
+import logging
+
+import werkzeug
+
+import odoo.http as http
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class HelpdeskTicketController(http.Controller):
+    @http.route("/ticket/close", type="http", auth="user")
+    def support_ticket_close(self, **kw):
+        """Close the support ticket"""
+        values = {}
+        for field_name, field_value in kw.items():
+            if field_name.endswith("_id"):
+                values[field_name] = int(field_value)
+            else:
+                values[field_name] = field_value
+        ticket = (
+            http.request.env["helpdesk.ticket"]
+            .with_user()
+            .search([("id", "=", values["ticket_id"])])
+        )
+        ticket.stage_id = values.get("stage_id")
+
+        return werkzeug.utils.redirect("/my/ticket/" + str(ticket.id))
+
+    @http.route("/new/ticket", type="http", auth="user", website=True)
+    def create_new_ticket(self, **kw):
+        categories = http.request.env["helpdesk.ticket.category"].search(
+            [("active", "=", True)]
+        )
+        email = http.request.env.user.email
+        name = http.request.env.user.name
+        return http.request.render(
+            "helpdesk_mgmt.portal_create_ticket",
+            {"categories": categories, "email": email, "name": name},
+        )
+
+    @http.route("/submitted/ticket", type="http", auth="user", website=True, csrf=True)
+    def submit_ticket(self, **kw):
+        vals = {
+            "partner_name": kw.get("name"),
+            "company_id": http.request.env.user.company_id.id,
+            "category_id": kw.get("category"),
+            "partner_email": kw.get("email"),
+            "description": kw.get("description"),
+            "name": kw.get("subject"),
+            "attachment_ids": False,
+            "channel_id": request.env["helpdesk.ticket.channel"]
+            .with_user()
+            .search([("name", "=", "Web")])
+            .id,
+            "partner_id": request.env["res.partner"]
+            .with_user()
+            .search([("name", "=", kw.get("name")), ("email", "=", kw.get("email"))])
+            .id,
+        }
+        new_ticket = request.env["helpdesk.ticket"].with_user().create(vals)
+        new_ticket.message_subscribe(partner_ids=request.env.user.partner_id.ids)
+        if kw.get("attachment"):
+            for c_file in request.httprequest.files.getlist("attachment"):
+                data = c_file.read()
+                if c_file.filename:
+                    request.env["ir.attachment"].with_user().create(
```

Comment:
@cubells This misunderstanding between you and @luffah is a bit odd.

Original line from v12: https://github.com/OCA/helpdesk/blob/2810cae768b85df5f7719e93d5d89ff961e0211e/helpdesk_mgmt/controllers/main.py#L62

You see, in this line you changed `sudo()` to `with_user()` which might not be correct.
If you read core code on how `with_user()` works, you'll see that when no `user` argument is given, it simply does nothing:
https://github.com/odoo/odoo/blob/84a6c2a59c59b88b0ad7b1f9112d331fa2a505c5/odoo/models.py#L5051-L5060

Now if the intention if this line (and similar ones) was to avoid access right restrictions, then it should have stayed as `sudo()`.

Reply:
Yes, I think @luffah is mixing guidelines. If you had in v12, `sudo(user)`, this is what must be replaced by `with_user(user)`, but `sudo()` without arguments must remain the same.

Reply:
@pedrobaeza 

thanks.

Reply:
Ok .... i ll wait this first PR to be validate and send correction if needed later.

Reply:
@gaikaz indeed, we need to avoid access right restrictions to allow portal users to create / see helpdesk tickets